### PR TITLE
magento/devdocs#: Add examples of hard and soft module dependencies

### PR DESCRIPTION
**Pull request by atwixfirster**
## Purpose of this pull request

This pull request (PR) adds examples of `hard` and `soft` module dependencies.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.html
- https://devdocs.magento.com/guides/v2.3/architecture/archi_perspectives/components/modules/mod_depend.html

## Links to Magento source code

- [hard](https://github.com/magento/magento2/blob/2.3.2/app/code/Magento/Cms/composer.json#L10)
- [soft](https://github.com/magento/magento2/blob/2.3.2/app/code/Magento/Cms/etc/module.xml#L11)

Thank you!

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.md [hunk 1]
```diff
@@ -31,13 +31,45 @@ A module with a *hard dependency* on another module cannot function without the
 * Deserializes an object declared in another module.
 * Uses or modifies the database tables used by another module.
 
+Module hard dependency defines in the `require` section of `app/code/<Vendor>/<Module>/composer.json` file. For example, :
```
Comment (rogyar) [inline guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.md]: Please adjust this part to make it more clear
```suggestion
Module hard dependency is defined in the `require` section of `app/code/<Vendor>/<Module>/composer.json` file. For example, :
```

File: guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.md [hunk 2]
```diff
@@ -31,13 +31,45 @@ A module with a *hard dependency* on another module cannot function without the
...
 ### Soft dependencies
 
 A module with a *soft dependency* on another module can function properly without the other module, even if it has a dependency on the other module. These modules:
 
 * Directly check another module's availability.
 * Extend another module's configuration.
-* Extend another module's {% glossarytooltip 73ab5daa-5857-4039-97df-11269b626134 %}layout{% endglossarytooltip %}.
+* Extend another module's [layout](https://glossary.magento.com/layout).
+
+Module hard dependency defines in the `<sequence>` section of `app/code/<Vendor>/<Module>/etc/module.xml` file. For example, :
```
Comment (rogyar) [inline guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.md]: Please adjust this part to make it more clear
```suggestion
Module soft dependency is defined in the `<sequence>` section of `app/code/<Vendor>/<Module>/etc/module.xml` file. For example, :
```

We are talking about soft dependency here, right? :)

File: guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.md (hunk 1 above)
Comment (hguthrie) [reply guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.md]: Do not make this passive. Is defined is not an improvement.

Comment (hguthrie) [inline guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.md]: Perhaps like this:

```suggestion
The `require` section of `app/code/<Vendor>/<Module>/composer.json` file contains hard dependency definitions for the module. For example, :
```

Comment (atwixfirster) [reply guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.md]: @hguthrie , thank you!

**implemented**

File: guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.md (hunk 2 above)
Comment (atwixfirster) [reply guides/v2.2/architecture/archi_perspectives/components/modules/mod_depend.md]: Golden eye, @rogyar 👍 

Many thanks

**Pull request merged**